src/SprykerEco/Zed/AmazonQuicksight/Business/AmazonQuicksightFacadeInterface.php

```diff
@@ -71,4 +96,37 @@ public function expandAnalyticsCollectionWithQuicksightAnalytics(
         AnalyticsRequestTransfer $analyticsRequestTransfer,
         AnalyticsCollectionTransfer $analyticsCollectionTransfer
     ): AnalyticsCollectionTransfer;
+
+    /**
+     * Specification:
+     * - Sends request to AWS API to get list of registered quicksight users. For more information see {@link https://docs.aws.amazon.com/quicksight/latest/APIReference/API_ListUsers.html}.
+     * - Filters out quicksight users with unsupported roles using {@link \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig::getQuicksightUserRoles()}.
+     * - Fetches user transfers from persistence.
+     * - Filters out user transfers with persisted quicksight user.
+     * - Matches registered quicksight users with users by username.
+     * - Persists matched quicksight users.
+     * - Adds errors to `QuicksightUserCollectionResponseTransfer.errors` if any occurs.
+     * - Returns `QuicksightUserCollectionResponseTransfer` with persisted quicksight users and errors if any occurs.
+     *
+     * @api
+     *
+     * @return \Generated\Shared\Transfer\QuicksightUserCollectionResponseTransfer
+     */
+    public function createQuicksightUsersForRegisteredQuicksightUsersMatchedExistingUsers(): QuicksightUserCollectionResponseTransfer;
```

Comment:
How about createMatchedQuicksightUsers? For now the method name is huge and sounds strange